src/pages/products/edit/ProductFields.tsx

```diff
@@ -9,63 +9,89 @@
  */
 
 import { Card, Element } from '@invoiceninja/cards';
-import { endpoint } from 'common/helpers';
+import { enterprisePlan } from 'common/guards/guards/enterprise-plan';
+import { proPlan } from 'common/guards/guards/pro-plan';
+import { endpoint, isHosted } from 'common/helpers';
 import { request } from 'common/helpers/request';
+import { toast } from 'common/helpers/toast/toast';
 import { useHandleCustomFieldChange } from 'common/hooks/useHandleCustomFieldChange';
 import { useInjectCompanyChanges } from 'common/hooks/useInjectCompanyChanges';
 import { updateRecord } from 'common/stores/slices/company-users';
+import { CustomFieldsPlanAlert } from 'components/CustomFieldsPlanAlert';
 import { Field } from 'pages/settings/custom-fields/components';
-import toast from 'react-hot-toast';
+import { FormEvent } from 'react';
 import { useTranslation } from 'react-i18next';
 import { useDispatch } from 'react-redux';
 
 export function ProductFields() {
-  const company = useInjectCompanyChanges();
+  const [t] = useTranslation();
+
   const dispatch = useDispatch();
 
-  const [t] = useTranslation();
+  const company = useInjectCompanyChanges();
+
   const handleCustomFieldChange = useHandleCustomFieldChange();
 
-  const onSave = (event: React.FormEvent<HTMLFormElement>) => {
+  const disabledCustomFields = !proPlan() && !enterprisePlan() && isHosted();
```

Comment:
Looks like we are copying the logic for disabling custom fields each time.
Can we move this into a hook, so we have logic centralized?

Reply:
Moved